server/server.go

```diff
@@ -2234,6 +2188,52 @@ func (s *StanServer) connectCB(m *nats.Msg) {
 	s.finishConnectRequest(client, req, m.Reply)
 }
 
+func (s *StanServer) processConnect(req *pb.ConnectRequest) (*client, bool, error) {
```

Comment:
This is not enough in a follower. It needs to handle possible existing client and replace with new one if old is not replying, or we need to change the way the leader replicates. I have added a test to the PR that shows the problem.